src/payment/models/buy-crypto/services/buy-crypto-notification.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable } from '@nestjs/common';
+import { MailService } from 'src/shared/services/mail.service';
+import { IsNull, Not } from 'typeorm';
+import { BuyCryptoRepository } from '../repositories/buy-crypto.repository';
+import { BuyCryptoBatchStatus } from '../entities/buy-crypto-batch.entity';
+
+@Injectable()
+export class BuyCryptoNotificationService {
+  constructor(private readonly buyCryptoRepo: BuyCryptoRepository, private readonly mailService: MailService) {}
+
+  async sentNotificationMails(): Promise<void> {
+    try {
+      const txOutput = await this.buyCryptoRepo.find({
+        where: {
+          recipientMail: IsNull(),
+          mailSendDate: IsNull(),
+          txId: Not(IsNull()),
```

Comment:
Should we check for isComplete instead of txId?

Reply:
right, but rather added to the query in addition to txId check